### ISSUE TITLE
[BUG] Image upload does not start when HTTP Proxy is configured

### ISSUE THREAD

**Issue by bashofmann** [harvester/harvester#2436]
**Describe the bug**
When a HTTP Proxy is configured in harvester, the VM image upload does not work. The upload does not start and errors with a timeout after 60 seconds. There are no obvious related errors in the logs.
When the HTTP Proxy settings are removed from Harvester, the image upload works without a problem. The `no_proxy` setting contained the cluster internal IP range `10.0.0.0/8`.

**To Reproduce**
* Configure a HTTP Proxy in Harvester
* Upload a VM image from your local machine through the UI

**Expected behavior**
The VM image is successfully uploaded

**Environment**
 - Harvester ISO version: 1.0.2

**Comment by futuretea** [comment]
@bashofmann `cattle-system` and `longhorn-system` should be added to the `no_proxy`
eg:
```json
{
  "httpProxy": "http://192.168.5.238:12333",
  "httpsProxy": "http://192.168.5.238:12333",
  "noProxy": "longhorn-system,cattle-system"
}
```

**Comment by bashofmann** [comment]
@futuretea Shouldn't these be added to noProxy if necessary automatically by harvester?

According to the docs: https://docs.harvesterhci.io/v1.0/settings/settings/#http-proxy 

> Harvester appends necessary addresses to user configured no-proxy to ensure the internal traffic works. i.e., localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,cattle-system.svc,.svc,.cluster.local

**Comment by TachunLin** [comment]
Verified **fixed** on v1.0-483ee6b3-head (7/21). Close this issue.

### Result
After the HTTP proxy is configured, we can upload image from local file correctly without error
  ![image](https://user-images.githubusercontent.com/29251855/180394751-7c764a98-b83b-4381-b9dd-7181e8eab90a.png)
  ![image](https://user-images.githubusercontent.com/29251855/180396275-b9c7f187-0b66-499a-aea9-2de66e4681c2.png)
  ![image](https://user-images.githubusercontent.com/29251855/180396398-ee4a4ee1-cb3b-4aaa-bbac-8a94e4ee96f6.png)


### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: v1.0-483ee6b3-head (7/21)

### Verify Steps
1. Clone ipxe-example vagrant project https://github.com/harvester/ipxe-examples
2. Edit settings.yml
3. `Set harvester_network_config.offline=true`
4. Create a one node air gapped Harvester with a HTTP proxy server 
5. Access Harvester settings page 
6. Add the following http proxy configuration
  ```
  {
    "httpProxy": "http://192.168.0.254:3128",
    "httpsProxy": "http://192.168.0.254:3128",
    "noProxy": "localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,cattle-system.svc,192.168.0.0/16,.svc,.cluster.local,example.com"
  }
  ```
  ![image](https://user-images.githubusercontent.com/29251855/180394751-7c764a98-b83b-4381-b9dd-7181e8eab90a.png)
7. Wait for 5 minutes for Harvester to apply the network setting
8. Go to images page
9. Upload image from local file

**Comment by TachunLin** [comment]
Verified **fixed** on master-1c060275-head (7/21). Close this issue.

### Result
After the HTTP proxy is configured, we can upload image from local file correctly without error
  ![image](https://user-images.githubusercontent.com/29251855/180407064-80c164ff-a46c-413d-ba8b-2a1bea023227.png)
  ![image](https://user-images.githubusercontent.com/29251855/180407214-0600cee9-28ce-46bd-b177-4f0151a5963b.png)


### Test Information
* Test Environment: 1 node harvester on local kvm machine
* Harvester version: master-1c060275-head (7/21)

### Verify Steps
1. Clone ipxe-example vagrant project https://github.com/harvester/ipxe-examples
2. Edit settings.yml
3. `Set harvester_network_config.offline=true`
4. Create a one node air gapped Harvester with a HTTP proxy server 
5. Access Harvester settings page 
6. Add the following http proxy configuration
  ```
  {
    "httpProxy": "http://192.168.0.254:3128",
    "httpsProxy": "http://192.168.0.254:3128",
    "noProxy": "localhost,127.0.0.1,0.0.0.0,10.0.0.0/8,cattle-system.svc,192.168.0.0/16,.svc,.cluster.local,example.com"
  }
  ```
  ![image](https://user-images.githubusercontent.com/29251855/180407430-bfe9140b-b0c1-44dc-9463-a478a6f705d3.png)

7. Wait for 5 minutes for Harvester to apply the network setting
8. Go to images page
9. Upload image from local file
